src/__stories__/button-group-story.tsx

```diff
@@ -10,51 +10,40 @@ export default {
 const handleOnPress = () => window.alert('pressed!');
 
 type Args = {
+    showPrimaryButton: boolean;
+    showSecondaryButton: boolean;
+    showLink: boolean;
     primaryButtonText: string;
     secondaryButtonText: string;
     linkText: string;
 };
 
-export const Default: StoryComponent<Args> = ({primaryButtonText, secondaryButtonText, linkText}) => {
-    const primaryButton = <ButtonPrimary onPress={handleOnPress}>{primaryButtonText}</ButtonPrimary>;
-
-    const secondaryButton = <ButtonSecondary onPress={handleOnPress}>{secondaryButtonText}</ButtonSecondary>;
-    const buttonLink = <ButtonLink onPress={handleOnPress}>{linkText}</ButtonLink>;
-    const longButton = (
-        <ButtonPrimary onPress={handleOnPress}>A very long action text in this button</ButtonPrimary>
-    );
-    return (
-        <Stack space={16}>
-            <Title1 as="h2">One Button</Title1>
-            <ButtonGroup primaryButton={primaryButton}></ButtonGroup>
-
-            <Title1 as="h2">Two Buttons</Title1>
-            <ButtonGroup primaryButton={primaryButton} secondaryButton={secondaryButton}></ButtonGroup>
-
-            <Title1 as="h2">Primary Button and Link</Title1>
-            <ButtonGroup primaryButton={primaryButton} link={buttonLink}></ButtonGroup>
-
-            <Title1 as="h2">Secondary Button and Link</Title1>
-            <ButtonGroup secondaryButton={secondaryButton} link={buttonLink}></ButtonGroup>
-
-            <Title1 as="h2">only Link</Title1>
-            <ButtonGroup link={buttonLink}></ButtonGroup>
-
-            <Title1 as="h2">Two Buttons and Link</Title1>
-            <ButtonGroup
-                primaryButton={primaryButton}
-                secondaryButton={secondaryButton}
-                link={buttonLink}
-            ></ButtonGroup>
-
-            <Title1 as="h2">Different-length Buttons</Title1>
-            <ButtonGroup primaryButton={longButton} secondaryButton={secondaryButton}></ButtonGroup>
-        </Stack>
-    );
+export const Default: StoryComponent<Args> = ({
+    showPrimaryButton,
+    showSecondaryButton,
+    showLink,
+    primaryButtonText,
+    secondaryButtonText,
+    linkText,
+}) => {
+    const primaryButton = showPrimaryButton ? (
+        <ButtonPrimary onPress={handleOnPress}>{primaryButtonText}</ButtonPrimary>
+    ) : undefined;
+
+    const secondaryButton = showSecondaryButton ? (
+        <ButtonSecondary onPress={handleOnPress}>{secondaryButtonText}</ButtonSecondary>
+    ) : undefined;
+
+    const buttonLink = showLink ? <ButtonLink onPress={handleOnPress}>{linkText}</ButtonLink> : undefined;
+
+    return <ButtonGroup primaryButton={primaryButton} secondaryButton={secondaryButton} link={buttonLink} />;
 };
 
 Default.storyName = 'Button group';
 Default.args = {
```

Comment:
you can hide primaryButtonText/secondaryButtonText/linkText controls when showPrimaryButton/showSecondaryButton/showLink are false:
https://storybook.js.org/docs/react/api/arg-types#if

```ts
Default.argTypes = {
    primaryButtonText: {if: {arg: 'showPrimaryButton'}},
    secondaryButtonText: {if: {arg: 'showSecondaryButton'}},
    linkText: {if: {arg: 'showLink'}},
};
```

Reply:
fixed